src/lib/proof-system/zkprogram.ts

```diff
@@ -804,8 +834,24 @@ function ZkProgram<
         programState.reset(methodIntfs[i].methodName);
       }
 
+      let publicOutput;
       let [publicOutputFields, proof] = MlPair.from(result);
-      let publicOutput = fromFieldConsts(publicOutputType, publicOutputFields);
+      if (publicInputAuxExists) {
+        publicInputAux = programState.getAuxilaryInput();
+        publicOutput = fromFieldAndAuxConsts(
+          publicOutputType,
+          publicOutputFields,
+          publicInputAux
+        );
+
+        programState.resetAuxCache();
+      } else {
+        // create public output when public input is pure
+        publicOutput = fromFieldAndAuxConsts(
```

Comment:
this is the pure case, we should be able to use the pure version of the function